### PR TITLE
Lpfg 682 list modules on course preview

### PR DESCRIPTION
**Pull request by mick-jones**
- Module listing added to course overview/preview with styling
- Added extra padding to top & bottom table rows and li elements
- Added /page/_course-overview.scss
- Renamed components to use camelCase for folder names
- courseEvents fleshed out and styled
- courseModules fleshed out and styled
- Removed isOptional name from optional checkbox and renamed to optional

### CODE REVIEW

File: views/component/courseModules/template.njk [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+                    <p class="govuk-body">{{ module.description }}</p>
+
+                    {% if module.optional === true %}
+                        <p class="optional">This module is optional</p>
+                    {% endif %}
+
+                    <a href="#" class="govuk-body govuk-link">Start</a>
+                </div>
+                <div class="govuk-grid-column-one-third">
+                    <p class="govuk-body type">{{ module.type }}</p>
```
Comment (m24murray) [inline views/component/courseModules/template.njk]: You should be passing to i18 to make sure your types are human readable

e.g. face-to-face > Face-to-face

File: views/component/courseModules/template.njk [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+                    <p class="govuk-body">{{ module.duration }}</p>
```
Comment (m24murray) [inline views/component/courseModules/template.njk]: We shouldn't be showing this value in its raw form. I don't know exactly how we do it in lpg-ui, but we should be converting seconds into its appropriate hours/minutes/seconds representation

File: views/page/course/module/add-module.html [hunk 3]
```diff
@@ -20,103 +20,93 @@
...
+                            {{ govukButton({
+                                text: "Add item",
+                                classes: "button-blue"
+                            }) }}
+                        </fieldset>
+                    </form>
+                <div class="button-container">
+                  <a class="govuk-button" href="#">Continue</a>          
+                </div>
+                <a class="cancel-button govuk-body govuk-link" href="#">Cancel</a>
```
Comment (m24murray) [inline views/page/course/module/add-module.html]: Can you sort out the value of this href? It should probs bring user back to course preview

File: views/component/courseModules/template.njk (hunk 1 above)
Comment (mick-jones) [reply views/component/courseModules/template.njk]: Fixed

File: package.json [hunk 4]
```diff
@@ -33,6 +33,7 @@
 		"app-root-path": "2.1.0",
 		"axios": "0.18.0",
 		"class-validator": "0.9.1",
+		"convert-seconds": "^1.0.1",
```
Comment (Peter-D-Taylor) [inline package.json]: The '^' should be removed, so that it uses a specific version

File: src/controllers/courseController.ts [hunk 5]
```diff
@@ -61,12 +62,19 @@ export class CourseController {
 			const faceToFaceModules = response.locals.course.modules.filter(
 				(module: Module) => module.type == Module.Type.FACE_TO_FACE
 			)
+			// console.log(Module.Type.FACE_TO_FACE)
```
Comment (Peter-D-Taylor) [inline src/controllers/courseController.ts]: comment can be removed

File: src/locale/en.json [hunk 6]
```diff
@@ -42,9 +45,7 @@
 	"validation.module.description.empty": "Description is required",
 	"validation_module_video_notFound": "There was a problem finding this video",
 	"validation_module_video_noDuration": "There was a problem finding the length of this video",
-    "validation_module_event_dateRanges_empty": "Start date is required",
-    "validation_module_event_dateRanges_past": "Start date must be in the future",
-    "validation_module_event_dateRanges_endBeforeStart": "Start time must be before end time",
-    "validation_module_event_date_invalid": "Start date is not valid",
```
Comment (Peter-D-Taylor) [inline src/locale/en.json]: Probably just missing from a merge, but this error message needs to stay

File: package.json [hunk 7]
```diff
@@ -41,10 +42,11 @@
 		"i18n-express": "1.1.3",
 		"is-valid-date": "0.0.1",
 		"log4js": "2.10.0",
-		"moment": "2.22.2",
+		"moment": "^2.22.2",
```
Comment (Peter-D-Taylor) [inline package.json]: same as above

**Pull request merged**